File: src/test/java/com/github/natche/gravatarjavaclient/profile/TokenSupplier.java
package com.github.natche.gravatarjavaclient.profile;

import java.nio.charset.StandardCharsets;

/**
 * A token supplier for grabbing the Profile API key from the system environment.
 */
public final class TokenSupplier {
    private static final String KEY_NAME = "GRAVATAR_JAVA_CLIENT_GITHUB_API_KEY";

    private static final GravatarProfileTokenProvider tokenSupplier = new GravatarProfileTokenProvider(
            () -> System.getenv(KEY_NAME).getBytes(StandardCharsets.US_ASCII), "TokenSupplier class"
    );

    /**
     * Suppress default constructor
     */
    private TokenSupplier() {
        throw new AssertionError("Cannot create instances of TokenSupplier");
    }

    /**
     * Returns the token supplier to grab the API key from the system environment.
     *
     * @return the token supplier to grab the API key from the system environment.
     */
    public static GravatarProfileTokenProvider getTokenSupplier() {
        return tokenSupplier;
    }
}

File: src/test/java/com/github/natche/gravatarjavaclient/profile/TokenSupplier.kt
package com.github.natche.gravatarjavaclient.profile

import java.nio.charset.StandardCharsets

/**
 * A token supplier for grabbing the Profile API key from the system environment.
 */
class TokenSupplier private constructor() {
    /**
     * Suppress default constructor
     */
    init {
        throw AssertionError("Cannot create instances of TokenSupplier")
    }

    companion object {
        private const val KEY_NAME = "GRAVATAR_JAVA_CLIENT_GITHUB_API_KEY"

        /**
         * Returns the token supplier to grab the API key from the system environment.
         *
         * @return the token supplier to grab the API key from the system environment.
         */
        val tokenSupplier = GravatarProfileTokenProvider(
            { System.getenv(KEY_NAME).toByteArray(StandardCharsets.US_ASCII) }, "TokenSupplier class"
        )
    }
}